components/datetime/src/external_loaders.rs

```diff
@@ -75,24 +75,33 @@ impl FixedDecimalFormatterLoader for ExternalLoaderCompiledData {
         &self,
         locale: &DataLocale,
         options: FixedDecimalFormatterOptions,
-    ) -> Result<FixedDecimalFormatter, DecimalError> {
-        FixedDecimalFormatter::try_new(locale, options)
+    ) -> Result<FixedDecimalFormatter, DataError> {
+        FixedDecimalFormatter::try_new(locale, options).map_err(|e| match e {
+            DecimalError::Data(e) => e,
+            _ => unreachable!(),
```

Comment:
this unreachable branch will disappear in 2.0, when the external constructor can return `DataError` only